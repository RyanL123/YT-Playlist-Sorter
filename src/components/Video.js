import React from "react";
import Typography from "@mui/material/Typography";
import { Box, Link, useMediaQuery } from "@mui/material";
import { convertISOtoString } from "../util/dateUtil";

const Video = (props) => {
    const mobile = useMediaQuery("(max-width:1000px)");
    return (
        <Box
            display="flex"
            flexDirection={mobile ? "column" : "row"}
            width="100%"
            justifyContent="space-between"
            my="20px"
        >
            <Box display="flex" flexDirection="column">
                <Typography variant="h5" color="text.primary">
                    <Link href={props.stats.link} target="_blank">
                        {props.stats.title}
                    </Link>
                </Typography>
                <Typography variant="h6" color="text.primary">
                    {props.stats.channel}
                </Typography>
                <Typography variant="h6" color="text.primary">
                    Views:{" "}
                    {props.stats.views
                        .toString()
                        .replace(/\B(?=(\d{3})+(?!\d))/g, ",")}
                </Typography>
                <Typography variant="h6" color="text.primary">
                    Likes:{" "}
                    {props.stats.likes
                        .toString()
                        .replace(/\B(?=(\d{3})+(?!\d))/g, ",")}
                </Typography>
                <Typography variant="h6" color="text.primary">
                    Upload Date: {convertISOtoString(props.stats.uploadDate)}
                </Typography>
                <Typography variant="h6" color="text.primary">
                    Duration:{" "}
                    {new Date(props.stats.duration * 1000)
                        .toISOString()
                        .substr(11, 8)}
                </Typography>
            </Box>
            <Box>
                <Link href={props.stats.link} target="_blank">
                    <img
                        src={props.stats.thumbnail}
                        style={{
                            width: mobile ? "100%" : "300px",
                            height: "auto",
                        }}
                        alt={props.stats.title}
                    />
                </Link>
            </Box>
        </Box>
    );
};

export default Video;
