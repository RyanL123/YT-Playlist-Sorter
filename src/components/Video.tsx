import { Box, Link, useMediaQuery } from "@mui/material";
import Typography from "@mui/material/Typography";
import * as React from "react";
import { IVideoMetadata } from "../types";
import { convertISOtoString } from "../util/dateUtil";

const Video = ({ metadata }: { metadata: IVideoMetadata }) => {
  const mobile = useMediaQuery("(max-width:1000px)");
  return (
    <Box
      display="flex"
      flexDirection={mobile ? "column" : "row"}
      width="100%"
      justifyContent="space-between"
      my="20px"
    >
      <Box display="flex" flexDirection="column">
        <Typography variant="h5" color="text.primary">
          <Link href={metadata.link} target="_blank">
            {metadata.title}
          </Link>
        </Typography>
        <Typography variant="h6" color="text.primary">
          {metadata.channel}
        </Typography>
        <Typography variant="h6" color="text.primary">
          Views:{" "}
          {metadata.views.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",")}
        </Typography>
        <Typography variant="h6" color="text.primary">
          Likes:{" "}
          {metadata.likes.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",")}
        </Typography>
        <Typography variant="h6" color="text.primary">
          Upload Date: {convertISOtoString(metadata.uploadDate)}
        </Typography>
        <Typography variant="h6" color="text.primary">
          Duration:{" "}
          {new Date(metadata.duration * 1000).toISOString().substr(11, 8)}
        </Typography>
      </Box>
      <Box>
        <Link href={metadata.link} target="_blank">
          <img
            src={metadata.thumbnail}
            style={{
              width: mobile ? "100%" : "300px",
              height: "auto",
            }}
            alt={metadata.title}
          />
        </Link>
      </Box>
    </Box>
  );
};

export default Video;
