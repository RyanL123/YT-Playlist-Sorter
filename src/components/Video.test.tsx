import React from "react";
import renderer from "react-test-renderer";
import Video from "./Video";

it("renders correctly", () => {
  const component = renderer.create(
    <Video
      metadata={{
        id: "123",
        statistics: {
          viewCount: "12345",
          likeCount: "12345",
        },
        snippet: {
          publishedAt: "1990-01-01",
          channelTitle: "Test",
          title: "Test",
          thumbnails: {
            medium: {
              url: "abc.com",
            },
          },
        },
        contentDetails: {
          duration: "PT1H1M1S",
        },
      }}
    />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
