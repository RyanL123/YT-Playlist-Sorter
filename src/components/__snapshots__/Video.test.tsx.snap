// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-18tsnoh"
>
  <div
    className="MuiBox-root css-j7qwjs"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-1xszx61-MuiTypography-root"
    >
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
        href="https://www.youtube.com/watch?v=123"
        onBlur={[Function]}
        onFocus={[Function]}
        target="_blank"
      >
        Test
      </a>
    </h5>
    <h6
      className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
    >
      Test
    </h6>
    <h6
      className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
    >
      Views:
       
      12,345
    </h6>
    <h6
      className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
    >
      Likes:
       
      12,345
    </h6>
    <h6
      className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
    >
      Upload Date:
       
      December 31, 1989
    </h6>
    <h6
      className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
    >
      Duration:
       
      01:01:01
    </h6>
  </div>
  <div
    className="MuiBox-root css-0"
  >
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
      href="https://www.youtube.com/watch?v=123"
      onBlur={[Function]}
      onFocus={[Function]}
      target="_blank"
    >
      <img
        alt="Test"
        src="abc.com"
        style={
          Object {
            "height": "auto",
            "width": "300px",
          }
        }
      />
    </a>
  </div>
</div>
`;
